### DISCUSSION TITLE
Allow to set PrivateMirror for ClamAV

### DISCUSSION THREAD

**Discussion by sotnikov-link** [nextcloud/all-in-one#2575] [Ideas] [answered]
I need to change ClamAV Update server. I want use PrivateMirror: https://docs.clamav.net/appendix/CvdPrivateMirror.html#use-freshclam-to-serve-only-whole-database-files-from-a-private-mirror

**Comment by szaimen** [comment]
Hi, why would you need such an option?

**Reply by sotnikov-link** [reply]
I need use PrivateMirror https://packages.microsoft.com/clamav/ because my internet provider block connection to https://database.clamav.net.

Some countries has restrictions too:
- https://check-host.net/check-http?host=https%3A%2F%2Fdatabase.clamav.net
- https://check-host.net/check-http?host=https%3A%2F%2Fpackages.microsoft.com%2Fclamav%2F

**Comment by szaimen** [comment]
> because my internet provider block connection to https://database.clamav.net.

This is what you should address by e.g. switchting the internet provider.

**Reply by sotnikov-link** [reply]
Or country…

**Reply by szaimen** [reply]
Are you in Vienna? I don't see why Vienna seems to be blocked. This should then rather be addressed by the clamav team...

**Reply by sotnikov-link** [reply]
I'm in Russia. ClamAV has option PrivateMirror to decide this problem. I wanted to know how to decide this problem more right. I see decision only to direct change freshclam.conf in docker container.

**Reply by sotnikov-link** [reply]
Maybe do you know other decisions?

**Reply by szaimen** [reply]
You could possibly get a VPS and run AIO there. Then clamav will work out of the box.

**Reply by szaimen** [reply]
And no, I dont know why they are blocking russia but I can guess.

**Reply by sotnikov-link** [reply]
Thanks for answers. I will try to change freshclam.conf in docker container.

**Reply by eagrzn** [reply]
@sotnikov-link, did you manage to change the config? I've been trying for several hours without success. The freshclam.conf file is not exposed in the volume, and I can't modify it in-place due to the read-only flag for the container.

**Reply by sotnikov-link** [reply]
@eagrzn sorry, I don't remember already. I didn't use it after experiments more 1 year ago.

**Reply by eagrzn** [reply]
Alright, thank you for responding so quickly, at least! :)

**Comment by szaimen** [comment]
Just FYI: I added this to https://github.com/nextcloud/all-in-one/issues/5251 now.

**Comment by szaimen** [comment, marked as answer]
Hi, we bundle the clamav database nowadays. Can you please check if this feature is still required?
